packages/framework/block-std/src/edgeless/model.ts

```diff
@@ -0,0 +1,205 @@
+   */
+  elementBound: Bound;
+  containedByBounds(bounds: Bound): boolean;
+  getNearestPoint(point: IVec): IVec;
+  intersectWithLine(start: IVec, end: IVec): PointLocation[] | null;
+  getRelativePointLocation(point: IVec): PointLocation;
+  hitTest(
+    x: number,
+    y: number,
+    options: IHitTestOptions,
+    host: EditorHost
+  ): boolean;
+  boxSelect(bound: Bound): boolean;
+}
+
+export class EdgelessBlockModel<
+    Props extends EdgelessSelectableProps = EdgelessSelectableProps,
+  >
+  extends BlockModel<Props>
+  implements IEdgelessElement
+{
+  private _externalXYWH: SerializedXYWH | undefined = undefined;
+
+  connectable = true;
+
+  rotate = 0;
+
+  boxSelect(bound: Bound): boolean {
+    return (
+      this.containedByBounds(bound) ||
+      bound.points.some((point, i, points) =>
+        this.intersectWithLine(point, points[(i + 1) % points.length])
+      )
+    );
+  }
+
+  containedByBounds(bounds: Bound): boolean {
+    const bound = Bound.deserialize(this.xywh);
+    const points = getPointsFromBoundsWithRotation({
+      x: bound.x,
+      y: bound.y,
+      w: bound.w,
+      h: bound.h,
+      rotate: this.rotate,
+    });
+    return points.some(point => bounds.containsPoint(point));
+  }
+
+  getNearestPoint(point: IVec): IVec {
+    const bound = Bound.deserialize(this.xywh);
+    return polygonNearestPoint(
+      rotatePoints(bound.points, bound.center, this.rotate ?? 0),
+      point
+    );
+  }
+
+  getRelativePointLocation(relativePoint: IVec): PointLocation {
+    const bound = Bound.deserialize(this.xywh);
+    const point = bound.getRelativePoint(relativePoint);
+    const rotatePoint = rotatePoints(
+      [point],
+      bound.center,
+      this.rotate ?? 0
+    )[0];
+    const points = rotatePoints(bound.points, bound.center, this.rotate ?? 0);
+    const tangent = polygonGetPointTangent(points, rotatePoint);
+
+    return new PointLocation(rotatePoint, tangent);
+  }
+
+  hitTest(x: number, y: number, _: IHitTestOptions, __: EditorHost): boolean {
+    const bound = Bound.deserialize(this.xywh);
+    return bound.isPointInBound([x, y], 0);
+  }
+
+  intersectWithLine(start: IVec, end: IVec): PointLocation[] | null {
+    const bound = Bound.deserialize(this.xywh);
+
+    return linePolygonIntersects(
+      start,
+      end,
+      rotatePoints(bound.points, bound.center, this.rotate ?? 0)
+    );
+  }
+
+  get elementBound() {
+    const bound = Bound.deserialize(this.xywh);
+    return Bound.from(getBoundsWithRotation({ ...bound, rotate: this.rotate }));
+  }
+
+  get externalBound(): Bound | null {
+    return this._externalXYWH ? Bound.deserialize(this._externalXYWH) : null;
+  }
+
+  get externalXYWH(): SerializedXYWH | undefined {
+    return this._externalXYWH;
+  }
+
+  set externalXYWH(xywh: SerializedXYWH | undefined) {
+    this._externalXYWH = xywh;
+  }
+
+  get group(): IEdgelessElement | null {
+    const surface = this.doc.getBlocksByFlavour('affine:surface')[0]?.model;
```

Comment:
This is very strange. If you really need to do something like this, you should add a flag on the block model instead of get a specific model flavour.

Reply:
```ts
class SurfaceModel extends BlockModel {
  [edgelessSurfaceRoot]: true
}

const surface = Object.values(this.doc.getBlocks()).find(...)
```